src/charm.py

```diff
@@ -1,104 +1,453 @@
+
+    def _configure_pebble(self, restart=False) -> None:
+        """Configure the Pebble layer.
+
+        Args:
+            restart (bool): Whether to restart the CU container.
+        """
+        plan = self._container.get_plan()
+        if plan.services != self._cu_pebble_layer.services:
+            self._container.add_layer(
+                self._container_name, self._cu_pebble_layer, combine=True
+            )
+            self._container.replan()
+            logger.info("New layer added: %s", self._cu_pebble_layer)
+        if restart:
+            self._container.restart(self._service_name)
+            logger.info("Restarted container %s", self._service_name)
+            return
+        self._container.replan()
+
+    def _update_fiveg_gnb_identity_relation_data(self) -> None:
+        """Publish GNB name and TAC in the `fiveg_gnb_identity` relation data bag."""
+        if not self.unit.is_leader():
+            return
+        fiveg_gnb_identity_relations = self.model.relations.get(GNB_IDENTITY_RELATION_NAME)
+        if not fiveg_gnb_identity_relations:
+            logger.info("No %s relations found.", GNB_IDENTITY_RELATION_NAME)
+            return
+
+        tac = self._charm_config.tac
+        if not tac:
+            logger.error(
+                "TAC value cannot be published on the %s relation", GNB_IDENTITY_RELATION_NAME
+            )
+            return
+        for gnb_identity_relation in fiveg_gnb_identity_relations:
+            self._gnb_identity_provider.publish_gnb_identity_information(
+                relation_id=gnb_identity_relation.id, gnb_name=self._gnb_name, tac=tac
+            )
+
+    def _update_fiveg_f1_relation_data(self) -> None:
+        """Publish F1 interface information in the `fiveg_f1` relation data bag."""
+        if not self.unit.is_leader():
+            return
+        fiveg_f1_relations = self.model.relations.get(F1_RELATION_NAME)
+        if not fiveg_f1_relations:
+            logger.info("No %s relations found.", F1_RELATION_NAME)
+            return
+        self._f1_provider.set_f1_information(
+            f1_ip_address=_get_pod_ip(), f1_port=self._charm_config.f1_port
+        )
+
+    @property
+    def _gnb_name(self) -> str:
+        """The gNB's name contains the model name and the app name.
+
+        Returns:
+            str: the gNB's name.
+        """
+        return f"{self.model.name}-{self.app.name}"
+
+    @property
+    def _cu_pebble_layer(self) -> Layer:
+        """Return pebble layer for the amf container.
+
+        Returns:
+            Layer: Pebble Layer
+        """
+        return Layer(
+            {
+                "services": {
+                    self._service_name: {
+                        "override": "replace",
+                        "startup": "enabled",
+                        "command": f"/opt/oai-gnb/bin/nr-softmodem -O {BASE_CONFIG_PATH}/{CONFIG_FILE_NAME} --sa --log_config.global_log_options level,nocolor,time",  # noqa: E501
```

Comment:
This is a question more than anything but I don't understand the part about the log level here.

Reply:
If you mean why are the logging options here and not in the config file, the answer is I just blindly copied the command from the upstream.
Moved that config to the config file.

Reply:
My question is really " what logging level are we setting?"

Reply:
Oh :D The log level is set in the `cu.conf`. The default value is `info`. This config here is used to format the message I believe.